### PR TITLE
SB-25735 - consumption - Added kong-api entries for private read api's of content, collection, question, questionset and search

### PR DESCRIPTION
**Pull request by shiva-rakshith**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-25735" title="SB-25735" target="_blank">EpicSB-25735</a>  Content Visibility - Allowing Creation, editing, and accessing(active/live content) of Private resource content on Diksha
  </td></table>
  <br />
 

…t and search

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/roles/kong-api/defaults/main.yml [hunk 1]
```diff
@@ -354,6 +354,25 @@ kong_apis:
...
+    upstream_url: "{{ knowledge_mw_service_url }}/v3/private/search"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist: 
+        - 'contentAdmin'
+        - 'contentTempAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:368]: Can you please explain why would we need two ACLs for this API?

File: ansible/roles/kong-api/defaults/main.yml [hunk 2]
```diff
@@ -354,6 +354,25 @@ kong_apis:
...
+    - name: rate-limiting
+      config.policy: local
+      config.hour: "{{ premium_consumer_large_rate_limit_per_hour }}"
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:371]: Please state why we need a premium rate limit for this? Who are the end users? How many calls can happen over an hour?

File: ansible/roles/kong-api/defaults/main.yml [hunk 3]
```diff
@@ -2212,6 +2231,24 @@ kong_apis:
...
+    uris: "{{ content_prefix }}/v1/private/read"
+    upstream_url: "{{ content_service_url }}/content/v4/private/read"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'anonymousContentAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:2244]: Is this API accessed by users / applications without the user logging in? What private data would an anonymous user read?

File: ansible/roles/kong-api/defaults/main.yml [hunk 4]
```diff
@@ -4834,6 +4871,24 @@ kong_apis:
...
+    uris: "{{ collection_prefix }}/v1/private/read"
+    upstream_url: "{{ content_service_url }}/collection/v4/private/read"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'anonymousContentAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:4884]: Is this API accessed by users / applications without the user logging in? What private data would an anonymous user read?

File: ansible/roles/kong-api/defaults/main.yml [hunk 5]
```diff
@@ -5515,6 +5570,24 @@ kong_apis:
...
+    uris: "{{ question_prefix }}/v1/private/read"
+    upstream_url: "{{ assessment_service_url }}/question/v4/private/read"
+    strip_uri: true
+    plugins:
+      - name: jwt
+      - name: cors
+      - "{{ statsd_pulgin }}"
+      - name: acl
+        config.whitelist:
+          - 'anonymousContentAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:5583]: Is this API accessed by users / applications without the user logging in? What private data would an anonymous user read?

File: ansible/roles/kong-api/defaults/main.yml [hunk 6]
```diff
@@ -5631,6 +5704,24 @@ kong_apis:
...
+    uris: "{{ questionset_prefix }}/v1/private/read"
+    upstream_url: "{{ assessment_service_url }}/questionset/v4/private/read"
+    strip_uri: true
+    plugins:
+      - name: jwt
+      - name: cors
+      - "{{ statsd_pulgin }}"
+      - name: acl
+        config.whitelist:
+          - 'anonymousContentAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:5717]: Is this API accessed by users / applications without the user logging in? What private data would an anonymous user read?

**Comment by shiva-rakshith** [comment]
@keshavprasadms - I have made the discussed changes.

File: ansible/roles/kong-api/defaults/main.yml [hunk 7]
```diff
@@ -349,7 +349,25 @@ kong_apis:
         - 'contentTempAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:349]: @shiva-rakshith Please remove this ACL if mobile apps do not need access to this API

Comment (shiva-rakshith) [reply ansible/roles/kong-api/defaults/main.yml:349]: For the existing api, we are not sure. But for the new api mobile apps do not need access.

File: ansible/roles/kong-api/defaults/main.yml [hunk 8]
```diff
@@ -5631,6 +5703,24 @@ kong_apis:
...
+      - name: jwt
+      - name: cors
+      - "{{ statsd_pulgin }}"
+      - name: acl
+        config.whitelist:
+          - 'contentAccess'
+      - name: rate-limiting
+        config.policy: local
+        config.hour: "{{ medium_rate_limit_per_hour }}"
+        config.limit_by: ip
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:5720]: Please use rate limiting by credential

File: ansible/roles/kong-api/defaults/main.yml [hunk 9]
```diff
@@ -5515,6 +5569,24 @@ kong_apis:
...
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:5586]: Please use rate limiting by credential

**Review by keshavprasadms — CHANGES_REQUESTED** [review]
@shiva-rakshith Please check my latest comments

File: ansible/roles/kong-api/defaults/main.yml (hunk 7 above)
Comment (keshavprasadms) [reply ansible/roles/kong-api/defaults/main.yml:349]: Ok, then we need to change this if you are unsure for now

Comment (shiva-rakshith) [reply ansible/roles/kong-api/defaults/main.yml:349]: Can we also revert the config.hour of existing api to 'premium_consumer_large_rate_limit_per_hour', we are not unsure if it disturbs anything.

File: ansible/roles/kong-api/defaults/main.yml (hunk 8 above)
Comment (shiva-rakshith) [reply ansible/roles/kong-api/defaults/main.yml:5720]: Updated suggested changes.

File: ansible/roles/kong-api/defaults/main.yml (hunk 9 above)
Comment (shiva-rakshith) [reply ansible/roles/kong-api/defaults/main.yml:5586]: Updated suggested changes.

File: ansible/roles/kong-api/defaults/main.yml (hunk 7 above)
Comment (keshavprasadms) [reply ansible/roles/kong-api/defaults/main.yml:349]: No, we will not revert that. We do not want to use premimu rate limits unless we are 100% sure about it.

**Review by keshavprasadms — APPROVED** [review]

File: ansible/roles/kong-api/defaults/main.yml (hunk 7 above)
Comment (shiva-rakshith) [reply ansible/roles/kong-api/defaults/main.yml:349]: @maheshkumargangula - For new PrivateSearch API, we are using config.hour as 'medium_rate_limit_per_hour'. @keshavprasadms has suggested to use the same config.hour for existing search API. Can we change from 'premium_consumer_large_rate_limit_per_hour' to 'medium_rate_limit_per_hour' for existing search API. Please confirm.

**Pull request merged**